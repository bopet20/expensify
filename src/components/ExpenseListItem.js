import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <p className="list-item__title">{description}</p>
      <span className="list-item__subtitle">{moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <p className="list-item__data">{numeral(amount/100).format('$0,0.00')}</p>
  </Link>
)

export default ExpenseListItem