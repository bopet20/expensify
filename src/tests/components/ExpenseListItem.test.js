import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

describe('<ExpenseListItem />', () => {
  it('renders with fixture data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot()
  })
})