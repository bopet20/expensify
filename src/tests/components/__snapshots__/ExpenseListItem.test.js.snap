// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListItem /> renders with fixture data 1`] = `
<Fragment>
  <Link
    replace={false}
    to="/edit/0"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.95
    - created
    December 31st, 1969
  </p>
</Fragment>
`;
